core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -639,19 +640,22 @@ private[spark] object Utils extends Logging {
       fs: FileSystem,
       conf: SparkConf,
       hadoopConf: Configuration,
-      fileOverwrite: Boolean): Unit = {
-    if (!targetDir.mkdir()) {
+      fileOverwrite: Boolean,
+      filename: Option[String] = None): Unit = {
```

Comment:
`doFetchFile` actually passes `filename` regardless of whether it's a file or a directory, and the code here treats both equally as far as `filename` is concerned, so I think that comment would be misleading.